python/ray/serve/_private/deployment_scheduler.py

```diff
@@ -338,11 +429,43 @@ def schedule(
                     replica_scheduling_request.replica_name
                 ] = replica_scheduling_request
 
-        for deployment_id, pending_replicas in self._pending_replicas.items():
-            if not pending_replicas:
-                continue
+        non_strict_pack_pgs_exist = any(
+            d.is_non_strict_pack_pg() for d in self._deployments.values()
+        )
+        # Schedule replicas using compact strategy.
+        if RAY_SERVE_USE_COMPACT_SCHEDULING_STRATEGY and not non_strict_pack_pgs_exist:
+
+            def key(req: ReplicaSchedulingRequest):
+                if (
+                    req.placement_group_bundles
+                    and req.placement_group_strategy == "STRICT_PACK"
+                ):
+                    return sum(
+                        [Resources(bundle) for bundle in req.placement_group_bundles],
+                        Resources(),
+                    )
+                else:
+                    return Resources(req.actor_resources)
+
+            # Flatten dict of deployment replicas into all replicas,
+            # then sort by decreasing resource size
+            replica_scheduling_requests = [
+                scheduling_request
+                for replicas in self._pending_replicas.values()
+                for scheduling_request in replicas.values()
+            ]
+
+            replica_scheduling_requests.sort(key=key, reverse=True)
+
+            # Schedule each replica
+            for scheduling_request in replica_scheduling_requests:
+                self._schedule_replica_compactly(scheduling_request)
```

Comment:
can we put this into a method like `_schedule_spread_deployment` to match `_schedule_compact_deployment`?